src/rust/sdk/src/tx_builder.rs

```diff
@@ -0,0 +1,560 @@
+    }
+
+    pub fn register_for_coin(mut self, coin: &TypeTag) -> Self {
+        let Ok(id) = MoveModuleId::from_str("0x1::managed_coin") else {
+            self.entry = Some(Err(anyhow!("invalid module id")));
+            return self
+        };
+
+        let entry = EntryFunction::new(
+            ModuleId::from(id),
+            ident_str!("register").to_owned(),
+            vec![coin.clone()],
+            vec![],
+        );
+
+        self.entry = Some(Ok(entry));
+        self
+    }
+
+    // Incentives entry functions
+
+    pub fn update_incentives(
+        mut self,
+        utility_coin: &TypeTag,
+        market_registration_fee: u64,
+        underwriter_registration_fee: u64,
+        custodian_registration_fee: u64,
+        taker_fee_divisor: u64,
+        integrator_fee_store_tiers: Vec<Vec<u64>>,
+    ) -> Self {
+        let entry: Result<EntryFunction> = (|| {
+            Ok(EntryFunction::new(
+                ModuleId::from(self.client.econia_module().to_owned()),
+                ident_str!("update_incentives").to_owned(),
+                vec![utility_coin.clone()],
+                vec![
+                    bcs::to_bytes(&market_registration_fee)?,
+                    bcs::to_bytes(&underwriter_registration_fee)?,
+                    bcs::to_bytes(&custodian_registration_fee)?,
+                    bcs::to_bytes(&taker_fee_divisor)?,
+                    bcs::to_bytes(&integrator_fee_store_tiers)?,
+                ],
+            ))
+        })();
+
+        self.entry = Some(entry);
+        self
+    }
+
+    pub fn upgrade_integrator_fee_store_via_coinstore(
+        mut self,
+        quote_coin: &TypeTag,
+        utility_coin: &TypeTag,
+        market_id: u64,
+        new_tier: u8,
+    ) -> Self {
+        let entry: Result<EntryFunction> = (|| {
+            Ok(EntryFunction::new(
+                ModuleId::from(self.client.econia_module().to_owned()),
+                ident_str!("upgrade_integrator_fee_store_via_coinstore").to_owned(),
+                vec![quote_coin.clone(), utility_coin.clone()],
+                vec![bcs::to_bytes(&market_id)?, bcs::to_bytes(&new_tier)?],
+            ))
+        })();
+
+        self.entry = Some(entry);
+        self
+    }
+
+    pub fn withdraw_integrator_fees_via_coinstores(
+        mut self,
+        quote_coin: &TypeTag,
+        utility_coin: &TypeTag,
+        market_id: u64,
+    ) -> Self {
+        let entry: Result<EntryFunction> = (|| {
+            Ok(EntryFunction::new(
+                ModuleId::from(self.client.econia_module().to_owned()),
+                ident_str!("withdraw_integrator_fees_via_coinstores").to_owned(),
+                vec![quote_coin.clone(), utility_coin.clone()],
+                vec![bcs::to_bytes(&market_id)?],
+            ))
+        })();
+
+        self.entry = Some(entry);
+        self
+    }
+
+    // Market entry functions
+
+    pub fn cancel_all_orders_user(mut self, market_id: u64, side: bool) -> Self {
```

Comment:
Can we have `side` accept an `enum` that maps to the corresponding bool from the Move code?

Will be much more user friendly to have `side=Side::Ask`.